packages/firestore/src/local/local_documents_view.ts

```diff
@@ -218,7 +221,43 @@ export class LocalDocumentsView {
         );
       })
       .next(matchingMutationBatches => {
+        mutationBatches = matchingMutationBatches;
+        // It is possible that a PatchMutation can make a document a match to the query,
+        // but the version in `remoteDocumentCache` is not a match yet (waiting for server
+        // to ack). To handle this, we find all document keys affected by some`PatchMutation`s
+        // that are not in `result` yet, and back fill them via `remoteDocumentCache.getEntries`,
+        // otherwise those `PatchMutations` will be ignored because no base document can be found,
+        // and lead to missing result for the query.
+        let missingBaseDocEntriesForPatching = documentKeySet();
```

Comment:
Nit: the added code makes the function considerably longer. Perhaps refactor it out into a helper function? Something like:
```javascript
.next(matchingMutationBatches => {
  mutationBatches = matchingMutationBatches;
  // Comment without changes here (or move it into the function as well)
  return this.getDocsWithLocalPatches(transaction, query, matchingMutationBatches);
})
.next(baseDocsForPatching) {
```

Reply:
If possible, I'd like to keep it as-is. 

It is a part of the main logic of `getDocumentsMatchingCollectionQuery`, read would need to understand this block to understand the whole function. I think in this case, it justifies to have a longer method.

Reply:
I disagree, but I won't insist.

To me, it seems like the general flow of this function would be clearer if it only contained high-level steps. Currently, it delegates a lot of important logic to `remoteDocumentsCache`, and I would say the fact that some parts are hidden behind an abstraction is actually helpful to readability, because the reader can focus on a single concern at a time. However, once again, I don't feel too strongly about it; feel free to keep as is.